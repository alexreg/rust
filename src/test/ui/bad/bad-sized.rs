// ignore-tidy-linelength

trait Trait {}

pub fn main() {
    let x: Vec<dyn Trait + Sized> = Vec::new();
    //~^ ERROR only auto traits can be used as additional traits in a trait object [E0225]
    //~| ERROR the size for values of type `dyn std::marker::Sized` cannot be known at compilation time [E0277]
    //~| the trait `std::marker::Sized` cannot be made into an object [E0038]
    //~| ERROR the size for values of type `dyn std::marker::Sized` cannot be known at compilation time [E0277]
    //~| the trait `std::marker::Sized` cannot be made into an object [E0038]
}
