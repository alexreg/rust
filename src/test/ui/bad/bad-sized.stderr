error[E0225]: only auto traits can be used as additional traits in a trait object
  --> $DIR/bad-sized.rs:6:28
   |
LL |     let x: Vec<dyn Trait + Sized> = Vec::new();
   |                    -----   ^^^^^
   |                    |       |
   |                    |       additional non-auto trait
   |                    |       trait alias used in trait object type (additional use)
   |                    first non-auto trait
   |                    trait alias used in trait object type (first use)

error[E0277]: the size for values of type `dyn std::marker::Sized` cannot be known at compilation time
  --> $DIR/bad-sized.rs:6:12
   |
LL |     let x: Vec<dyn Trait + Sized> = Vec::new();
   |            ^^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `dyn std::marker::Sized`
   = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
   = note: required by `std::vec::Vec`

error[E0038]: the trait `std::marker::Sized` cannot be made into an object
  --> $DIR/bad-sized.rs:6:12
   |
LL |     let x: Vec<dyn Trait + Sized> = Vec::new();
   |            ^^^^^^^^^^^^^^^^^^^^^^ the trait `std::marker::Sized` cannot be made into an object
   |
   = note: the trait cannot require that `Self : Sized`

error[E0277]: the size for values of type `dyn std::marker::Sized` cannot be known at compilation time
  --> $DIR/bad-sized.rs:6:37
   |
LL |     let x: Vec<dyn Trait + Sized> = Vec::new();
   |                                     ^^^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `dyn std::marker::Sized`
   = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
   = note: required by `std::vec::Vec::<T>::new`

error[E0038]: the trait `std::marker::Sized` cannot be made into an object
  --> $DIR/bad-sized.rs:6:37
   |
LL |     let x: Vec<dyn Trait + Sized> = Vec::new();
   |                                     ^^^^^^^^ the trait `std::marker::Sized` cannot be made into an object
   |
   = note: the trait cannot require that `Self : Sized`

error: aborting due to 5 previous errors

Some errors have detailed explanations: E0038, E0225, E0277.
For more information about an error, try `rustc --explain E0038`.
